bin/update-tables.py

```diff
@@ -666,6 +668,47 @@ def filenames() -> list[str]:
         return [os.path.join(PATH_DATA, match.string) for match in filename_matches]
 
 
+def replace_if_modified(new_filename: str, original_filename: str) -> None:
+    """Replace original file with new file only if there are significant changes.
+
+    If only the 'This code generated' timestamp line differs, discard the new file.
+    If there are other changes or the original doesn't exist, replace it.
+    """
+    if os.path.exists(original_filename):
+        with open(original_filename, 'r', encoding='utf-8') as f1, \
+                open(new_filename, 'r', encoding='utf-8') as f2:
+            old_lines = f1.readlines()
+            new_lines = f2.readlines()
+
+        # Generate diff
+        diff_lines = list(difflib.unified_diff(old_lines, new_lines,
+                                               fromfile=original_filename,
+                                               tofile=new_filename,
+                                               lineterm=''))
+
+        # Check if only the 'This code generated' line is different
+        significant_changes = False
+        for line in diff_lines:
+            if (line.startswith('@@') or
```

Comment:
Re-readed and understood: https://docs.python.org/3/library/stdtypes.html#str.startswith

You can use `line.startswith(('@@', '---'))` and list all prefixes in the tuple instead of multiple `or`. Code will look cleaner.